src/waves/waves.js

```diff
@@ -34,7 +34,9 @@ export class MdWaves {
     }
 
     this.attributeManager.addClasses(classes);
-    Waves.attach(this.element);
+    if (!(typeof window.require === 'function' && typeof window.require.specified === 'function' && window.require.specified('waves'))) {
```

Comment:
In my project both window.require and window.requirejs is defined. I will add a commit in the branch and change it to window.requirejs so it can be tested more.